### ISSUE TITLE
Quests always hidden / combine markers

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#179]
When there are many quests in an area they are often hidden and cannot be shown all at a time. This means user easily "overlook" quests, which is not nice, especially if these quests are more important/"nicer" ones than all the house floor quests you may e.g. see in big cities… The opening hours quests or so may not be shown, because of all these other quests.

Currently the markers are just hidden:
```
+---+         +----+
|Q|2|         |Q|1?|
+---+         +----+


+---+        +-----+
|Q|1|        | Q|1?| <-- (actually here is one more quest, hidden)
+---+        +-----+
```

Now what I'd like to see would be a "combined" view of these open quests, when they cannot all be shown:
```
+---+         +----+
|Q|2|         |Q|1?|
+---+         +----+


+------------------+
|                  |
|  3 quests here!  |
|                  |
+------------------+
```

This way users know there are 3 quests at this area, which could be solved. So they can just scroll in to see them all…
If they are hidden, they would not even notice them and thus ignore them. Especially when scrolling in and out some quests may be shown and hidden, which can be very annoying, when you are trying to tap on one and then you accidentally scrolled out a bit.

**Comment by westnordost** [comment]
This is technically not possible (yet) with tangram-es. As far as I know, there is currently nothing in the works that would enable such a feature.

On 4 May 2017 09:23:19 CEST, rugk <notifications@github.com> wrote:
>When there are many quests in an area they are often hidden and cannot
>be shown all at a time. This means user easily "overlook" quests, which
>is not nice, especially if these quests are more important/"nicer" ones
>than all the house floor quests you may e.g. see in big cities… The
>opening hours quests or so may not be shown, because of all these other
>quests.
>
>Currently the markers are just hidden:
>```
>+---+         +----+
>|Q|2|         |Q|1?|
>+---+         +----+
>
>
>+---+        +-----+
>|Q|1|        | Q|1?| <-- (actually here is one more quest, hidden)
>+---+        +-----+
>```
>
>Now what I'd like to see would be a "combined" view of these open
>quests, when they cannot all be shown:
>```
>+---+         +----+
>|Q|2|         |Q|1?|
>+---+         +----+
>
>
>+------------------+
>|                  |
>|  3 quests here!  |
>|                  |
>+------------------+
>```
>
>This way users know there are 3 quests at this area, which could be
>solved. So they can just scroll in to see them all…
>If they are hidden, they would not even notice them and thus ignore
>them. Especially when scrolling in and out some quests may be shown and
>hidden, which can be very annoying, when you are trying to tap on one
>and then you accidentally scrolled out a bit.
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/179

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by rugk** [comment]
Mhh, could not you just replace (multiple) makers with one at a certain zoom level. So just remove some markers and add one instead of multiple ones?

**Comment by westnordost** [comment]
No.
Technically, all markers are at zoom level 14 (I think) and when loaded, are simply dumped onto the map. I do not load another different set of quests on different zoom levels like you would expect it to work with a tileserver (different tiles for different zoom levels). The whole thing - displaying all these markers on the map - is a bit of a hack anyway as doing exactly what I need is not really possible yet in the rendering library I use. I think it is a waste of time to try to hack this even more to get a certain behavior, as I hope that sooner or later, such a feature will be supported by the library itself. Until then, there are enough other things to do.

TLDR: I do like your idea and I imagine that this would be the best behavior, but it is not technically possible yet - not with justifiable effort at least, and that boils down to the same thing.

**Comment by rugk** [comment]
> I hope that sooner or later, such a feature will be supported by the library itself.

Well… being an open-source lib on GitHub, we can certainly do more than just hope.
Do you know an open issue for that feature or can we create a [new one](https://github.com/tangrams/tangram-es/issues)? (I hope I've linked correctly)

Hey, BTW seeing the screenshots I just got a new idea: https://github.com/westnordost/StreetComplete/issues/183

**Comment by escoand** [comment]
Don't know if this is the correct issue https://github.com/tangrams/tangram-es/issues/1222

**Comment by rugk** [comment]
Which would be bad…
